server/src/main/java/org/opensearch/indices/recovery/RemoteStoreReplicaRecoverySourceHandler.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.indices.recovery;
+
+import org.apache.lucene.index.IndexCommit;
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.StepListener;
+import org.opensearch.common.concurrent.GatedCloseable;
+import org.opensearch.common.lease.Releasable;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.index.engine.RecoveryEngineException;
+import org.opensearch.index.seqno.SequenceNumbers;
+import org.opensearch.index.shard.IndexShard;
+import org.opensearch.indices.RunUnderPrimaryPermit;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.Transports;
+
+import java.io.IOException;
+import java.util.function.Consumer;
+
+/**
+ * This handler is used when peer recovery target is a remote store enabled replica.
+ */
+public class RemoteStoreReplicaRecoverySourceHandler extends RecoverySourceHandler {
+
+    public RemoteStoreReplicaRecoverySourceHandler(
+        IndexShard shard,
+        RecoveryTargetHandler recoveryTarget,
+        ThreadPool threadPool,
+        StartRecoveryRequest request,
+        int fileChunkSizeInBytes,
+        int maxConcurrentFileChunks,
+        int maxConcurrentOperations
+    ) {
+        super(shard, recoveryTarget, threadPool, request, fileChunkSizeInBytes, maxConcurrentFileChunks, maxConcurrentOperations);
+    }
+
+    @Override
+    protected void innerRecoveryToTarget(ActionListener<RecoveryResponse> listener, Consumer<Exception> onFailure) throws IOException {
+        // A replica of an index with remote translog does not require the translogs locally and keeps receiving the
+        // updated segments file on refresh, flushes, and merges. We plan to make the existing replication call to
```

Comment:
Yes, the "we plan to" comment tells us what is coming as this is an incremental PR to the bigger effort. However, I get that we can make it shorter and concise. I have reworded the "we plan to" section. Let me know if this makes sense.

Reply:
The update looks good, thanks!

I know this is nit-picky but comments like "we plan to..." very often become stale and will still be present in the code even after all the planned work is done, because the future work may not touch this exact piece of code again. So unless there is a strong need to document future intent I find it is better to try to avoid it.